src/pyk/kcfg/show.py

```diff
@@ -325,20 +337,14 @@ def _hide_cells(_k: KInner) -> KInner:
 
             def to_rule(edge: KCFG.Edge, *, claim: bool = False) -> KRuleLike:
                 sentence_id = f'BASIC-BLOCK-{edge.source.id}-TO-{edge.target.id}'
-                init_cterm = CTerm(hide_cells(edge.source.cterm.config), ())
-                for c in edge.source.cterm.constraints:
-                    assert type(c) is KApply
-                    if c.label.name == '#Ceil':
-                        _LOGGER.warning(f'Ignoring Ceil condition: {c}')
-                    else:
-                        init_cterm.add_constraint(c)
-                target_cterm = CTerm(hide_cells(edge.target.cterm.config), ())
-                for c in edge.source.cterm.constraints:
-                    assert type(c) is KApply
-                    if c.label.name == '#Ceil':
-                        _LOGGER.warning(f'Ignoring Ceil condition: {c}')
-                    else:
-                        target_cterm.add_constraint(c)
+                init_constraints = [
+                    c for c in edge.source.cterm.constraints if not (type(c) is KApply and c.label.name == '#Ceil')
```

Comment:
```suggestion
                    c for c in edge.source.cterm.constraints if type(c) is not KApply or c.label.name != '#Ceil'
```

I personally prefer this variant, but the original is fine too.

Reply:
I factored out a helper.